Do not upper-bound the meson-python version. Allows building with meson-python 0.16.0.
https://github.com/numpy/numpy/commit/43f9f1b59ef64c7797816fcb7fa6925963fbdcf9
--- pyproject.toml
+++ pyproject.toml
@@ -2,7 +2,7 @@
 build-backend = "mesonpy"
 requires = [
     "Cython>=0.29.34,<3.1",
-    "meson-python>=0.15.0,<0.16.0",
+    "meson-python>=0.15.0",
 ]
 
 [project]
